test/dialects/spark3_grammar_test.py

```diff
@@ -0,0 +1,97 @@
+"""Test some specific spark3 grammars."""
```

Comment:
this was mainly for my learning/troubleshooting

Reply:
As far as I can see there's not really anything in this file that's not already being tested by the sql/yml tests so think we should remove this file when it comes to the non-draft PR.

(Nice work on the SQL tests btw, they look pretty thorough 😄)

Reply:
it was...a process

these aren't completely redundant since you can verify that a snippet matches more than one grammar, and because of the xfails

my plan is to make issues for the stuff that's still broken, note the issue numbers below, and then come back here and make the tests pass. how would you feel about leaving those in place? 

of course i can also just maintain this file on a branch but my thought is that it could trim the learning curve for e.g. other folks trying to get into spark3 development

Reply:
I would MUCH prefer these to be moved to SQL and YML files in `test/fixtures/dialects/spark3`.

There are several benefits to being in there:
- Easier to read and write
- We generate the parse tree in the YML file and commit this to git so can see any unexpected changes to this for future PRs - even if completely unrelated to this.
- We're testing whole SQL statements, rather than partial pieces, which might work individually but not together. In the real world we write SQL and not fragments.
- Allows us to run our [lint test](https://github.com/sqlfluff/sqlfluff/pull/2473).
- Deeper tests usually aren't required.

The downside is you can't be as specific in checks (it's a straight match or fail), so sometimes need Python tests for more complicated code, or python functions within segments. But they should not be used for SQL statements for above reasons.

Reply:
Oh and feel free to add commented out tests with TODOs in the SQLs for future PRs.

Reply:
Commented tests with TODOs work for me.

Do you have thoughts on how to make it more obvious how to write this kind of test when needed during development? I found myself needing the fine-grained test to understand what was going wrong and having to do a fair amount of reading to understand how to set it up. Of course I now know right where this commit is and can always dig it up, but just thinking of the next casual dialect developer. (Totally fine if not I have what I need to adjust this PR)

Reply:
I tend to start with small bits of SQL and then just use `sqlfluff parse` to look at the tree and comment/uncomment code until I get it to work. Not the most efficiint I agree but usually gets me there.

I do this in one of three ways:
- Command line for simple statements (`echo "select * from values 1, values 2;" | sqlfluff parse --dialect spark3 -`
- `test.sql` for more complex ones (`sqlfluff parse --dialect spark3 test.sql`). This is excluded from git so good place to test.
- From `test/fixtures` directory (test driven development! I should do this more).

That's usually sufficient for me but maybe some (particularly Python developers - which I am not), may prefer your approach.

Either way, there's good reason to move them to the standard ones, once you've figured it out as discussed above.

Reply:
Thanks, TDD out of the fixtures directory is approximately what I did until I was stumped by the parser's terse output. It worked well for most of what I've done so far, though I feel I found the limits while developing this PR. 

I'll noodle on how to take the pain/learning curve out of temporarily standing up very specific Python tests.